lib/Core/numberUtils.js

```diff
@@ -0,0 +1,34 @@
+'use strict';
+
+function isInteger(value) {
+    return !isNaN(value) && parseInt(Number(value), 10) === value && !isNaN(parseInt(value, 10));
+}
+
+function isNumeric(n) {
+    // isNumeric("15") = true
+    // isNumeric("-1.1") = true
+    // isNumeric("-1.1a") = false
+    // isNumeric("Infinity") = false
+    // isNumeric("NaN") = false
+    return !isNaN(parseFloat(n)) && isFinite(n);
+}
+
+function startsNumeric(n) {
+    // startsNumeric("15") = true
+    // startsNumeric("-1.1") = true
+    // startsNumeric("-1.1a") = true   // ** Note difference from isNumeric.
+    // startsNumeric("Infinity") = true  // ** Also different, but unlikely to be a problem.
+    // startsNumeric("NaN") = false
+    return !isNaN(parseFloat(n));
+}
+
+function compareStringsAndNumbers(a, b) {
```

Comment:
No, that one does lots of clever stuff.  For example, it sorts correctly when numbers are in the middle or at the end of the string.  It also handles dates I believe.